### PR TITLE
APPSRE-6609 OCMap for trigger-cleaner

### PR DESCRIPTION
**Pull request by fishi0x01**
New OCMap for `openshift-saas-deploy-trigger-cleaner` integration.

### CODE REVIEW

File: reconcile/typed_queries/tekton_pipeline_providers.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    query_func: Optional[Callable] = None,
+) -> list[PipelinesProviderTektonV1]:
+    if not query_func:
+        query_func = gql.get_api().query
+    providers: list[PipelinesProviderTektonV1] = []
+    for provider in query(query_func).pipelines_providers or []:
+        if not isinstance(provider, PipelinesProviderTektonV1):
+            continue
+        providers.append(provider)
+    return providers
```
Comment (hemslo) [inline reconcile/typed_queries/tekton_pipeline_providers.py:21]: This is equivalent to
```python
pipelines_providers = query(query_func).pipelines_providers or []
return [p for p in pipeline_providers if isinstance(p, PipelinesProviderTektonV1)]
```

Comment (fishi0x01) [reply reconcile/typed_queries/tekton_pipeline_providers.py:21]: Good catch!
The original intent of this is to convert the existing logic towards typed queries. I.e., Im more inclined to keep the original structure for now.

**Review by chassing — APPROVED** [review]
A minor nit, otherwise LGTM

File: reconcile/openshift_saas_deploy_trigger_cleaner.py [hunk 2]
```diff
@@ -39,53 +47,51 @@ def run(
...
         thread_pool_size=thread_pool_size,
     )
+
     if defer:
         defer(oc_map.cleanup)
 
-    for pp in pipelines_providers:
-        retention = pp.get("retention")
+    for pp in pipeline_providers:
+        retention = pp.retention
```
[the remark this answers is not in the document]
Comment (fishi0x01) [reply reconcile/openshift_saas_deploy_trigger_cleaner.py]: Added. Thanks!

**Pull request merged**